libfoundation/src/foundation-java-private.cpp

```diff
@@ -1125,38 +1169,258 @@ static bool MCJavaClassNameToPathString(MCNameRef p_class_name, MCStringRef& r_s
     return MCStringCopy(*t_escaped, r_string);
 }
 
-bool MCJavaPrivateCallJNIMethod(MCNameRef p_class_name, void *p_method_id, int p_call_type, MCTypeInfoRef p_signature, void *r_return, void **p_args, uindex_t p_arg_count)
+static jclass MCJavaPrivateFindClass(MCNameRef p_class_name)
 {
-    if (p_method_id == nullptr)
+    if (MCStringBeginsWith(MCNameGetString(p_class_name),
+                           MCSTR("com.runrev.android"),
+                           kMCStringOptionCompareExact))
+    {
+#if defined(TARGET_SUBPLATFORM_ANDROID)
```

Comment:
Perhaps it is worth adding a comment here to describe what we are doing and why (i.e. the whole ClassLoader context thing).